lib/instructionsets.ts

```diff
@@ -80,27 +84,29 @@ export class InstructionSets {
         };
     }
 
-    async getCompilerInstructionSetHint(compilerArch, exe) {
+    async getCompilerInstructionSetHint(compilerArch: string | boolean, exe: string): Promise<string> {
         return new Promise(resolve => {
-            if (compilerArch) {
-                _.each(this.supported, (method, instructionSet) => {
-                    for (let target of method.target) {
+            if (compilerArch && typeof compilerArch === 'string') {
+                for (const instructionSet in this.supported) {
+                    const method = this.supported[instructionSet];
+                    for (const target of method.target) {
```

Comment:
Cant comment on lines not edited:

Can you add returns after the resolves (94 & 103)? Although the promise status won't get updated after the first resolve/reject is issued, code still runs afaik - so we can save some loops if we stop when not needed

Reply:
hmm, I thought it would. I remember the linter complaining when I put code after a resolve. But will try